### DISCUSSION TITLE
Minisforum UM790 pro unable to make it work

### DISCUSSION THREAD

**Discussion by MMaiero** [isc30/ryzen-gpu-passthrough-proxmox#5] [Q&A] [answered]
Hello,
thanks for the great and in depth guide!
Unfortunately, I cannot make it work with my Minisforum UM790 Pro with AMD 7940HS.

In my Windows 11, I get the following:
<img width="1557" alt="image" src="https://github.com/isc30/ryzen-7000-series-proxmox/assets/10448126/2bea2eb7-b309-4e4a-a244-afe85a766c86">

<img width="803" alt="image" src="https://github.com/isc30/ryzen-7000-series-proxmox/assets/10448126/0f85e9d6-d782-4ff5-815c-4c7f487a4239">

I have first tried with the original 1.0.3 BIOS and later switched to the latest 1.0.7.
I have also updated to the latest AMD drivers (released on the 19th) but with the same result.

Connected via HDMI, the monitor shows a completely glitched image, almost unreadable.

I have thought that a possible issue could be related to a different AMDGop driver. I have tried to run the tools in the specific folder running them in the actual VM (I don't have other windows machines) but without being able to make them work.

Do you have any suggestion or guidance to make it working?

Thanks.

**Comment by MMaiero** [comment]
The error reported when starting:

<img width="414" alt="image" src="https://github.com/isc30/ryzen-7000-series-proxmox/assets/10448126/d2be49f6-326b-4fec-ab9b-6b36a2f9cc1c">

**Reply by isc30** [reply]
Hey, @MMaiero I had reports from other users with the same processor as you getting it to work, it's definitely possible 💯 .
Let's try to find the root cause, I would recommend going step by step:

1. ### Run the following commands and paste them into a reply to diagnose possible issues:
    
    1. `pveversion`
    2. `lspci -lvk`
    3. `cat /etc/default/grub`
    4. `cat /etc/modprobe.d/vfio.conf`
    3. `cat /etc/modprobe.d/pve-blacklist.conf`

1. ### Follow the guide installing a Windows10 system using SeaBIOS:

    If this works, it means that the extracted `vbios` is working correctly, and the issue is related to UEFI and AMDGopDriver.

I'm also available in Zulip from time to time: https://isc30.zulipchat.com/join/36je7erap524zo5hh2gjohyx/

**Reply by MMaiero** [reply]
Hi, thanks for the quick reply.

Regarding point 1, here are my outputs

```
root@pve:~# pveversion
perl: warning: Setting locale failed.
perl: warning: Please check that your locale settings:
	LANGUAGE = (unset),
	LC_ALL = (unset),
	LC_TERMINAL = "iTerm2",
	LC_CTYPE = "UTF-8",
	LANG = "en_US.UTF-8"
    are supported and installed on your system.
perl: warning: Falling back to a fallback locale ("en_US.UTF-8").
pve-manager/8.0.4/d258a813cfa6b390 (running kernel: 6.2.16-15-pve)
root@pve:~# lspci -nnk
00:00.0 Host bridge [0600]: Advanced Micro Devices, Inc. [AMD] Device [1022:14e8]
	Subsystem: Device [1f4c:b002]
00:00.2 IOMMU [0806]: Advanced Micro Devices, Inc. [AMD] Device [1022:14e9]
	Subsystem: Device [1f4c:b002]
00:01.0 Host bridge [0600]: Advanced Micro Devices, Inc. [AMD] Device [1022:14ea]
00:02.0 Host bridge [0600]: Advanced Micro Devices, Inc. [AMD] Device [1022:14ea]
00:02.1 PCI bridge [0604]: Advanced Micro Devices, Inc. [AMD] Device [1022:14ee]
	Subsystem: Device [1f4c:b002]
	Kernel driver in use: pcieport
00:02.3 PCI bridge [0604]: Advanced Micro Devices, Inc. [AMD] Device [1022:14ee]
	Subsystem: Device [1f4c:b002]
	Kernel driver in use: pcieport
00:02.4 PCI bridge [0604]: Advanced Micro Devices, Inc. [AMD] Device [1022:14ee]
	Subsystem: Device [1f4c:b002]
	Kernel driver in use: pcieport
00:03.0 Host bridge [0600]: Advanced Micro Devices, Inc. [AMD] Device [1022:14ea]
00:03.1 PCI bridge [0604]: Advanced Micro Devices, Inc. [AMD] Device [1022:14ef]
	Subsystem: Device [1f4c:b002]
	Kernel driver in use: pcieport
00:04.0 Host bridge [0600]: Advanced Micro Devices, Inc. [AMD] Device [1022:14ea]
00:04.1 PCI bridge [0604]: Advanced Micro Devices, Inc. [AMD] Device [1022:14ef]
	Subsystem: Device [1f4c:b002]
	Kernel driver in use: pcieport
00:08.0 Host bridge [0600]: Advanced Micro Devices, Inc. [AMD] Device [1022:14ea]
00:08.1 PCI bridge [0604]: Advanced Micro Devices, Inc. [AMD] Device [1022:14eb]
	Subsystem: Device [1f4c:b002]
	Kernel driver in use: pcieport
00:08.2 PCI bridge [0604]: Advanced Micro Devices, Inc. [AMD] Device [1022:14eb]
	Subsystem: Device [1f4c:b002]
	Kernel driver in use: pcieport
00:08.3 PCI bridge [0604]: Advanced Micro Devices, Inc. [AMD] Device [1022:14eb]
pcilib: Error reading /sys/bus/pci/devices/0000:00:08.3/label: Operation not permitted
	Subsystem: Device [1f4c:b002]
	Kernel driver in use: pcieport
00:14.0 SMBus [0c05]: Advanced Micro Devices, Inc. [AMD] FCH SMBus Controller [1022:790b] (rev 71)
	Subsystem: Device [1f4c:b002]
	Kernel driver in use: piix4_smbus
	Kernel modules: i2c_piix4, sp5100_tco
00:14.3 ISA bridge [0601]: Advanced Micro Devices, Inc. [AMD] FCH LPC Bridge [1022:790e] (rev 51)
	Subsystem: Device [1f4c:b002]
00:18.0 Host bridge [0600]: Advanced Micro Devices, Inc. [AMD] Device [1022:14f0]
00:18.1 Host bridge [0600]: Advanced Micro Devices, Inc. [AMD] Device [1022:14f1]
00:18.2 Host bridge [0600]: Advanced Micro Devices, Inc. [AMD] Device [1022:14f2]
00:18.3 Host bridge [0600]: Advanced Micro Devices, Inc. [AMD] Device [1022:14f3]
	Kernel driver in use: k10temp
	Kernel modules: k10temp
00:18.4 Host bridge [0600]: Advanced Micro Devices, Inc. [AMD] Device [1022:14f4]
00:18.5 Host bridge [0600]: Advanced Micro Devices, Inc. [AMD] Device [1022:14f5]
00:18.6 Host bridge [0600]: Advanced Micro Devices, Inc. [AMD] Device [1022:14f6]
00:18.7 Host bridge [0600]: Advanced Micro Devices, Inc. [AMD] Device [1022:14f7]
01:00.0 Ethernet controller [0200]: Realtek Semiconductor Co., Ltd. RTL8125 2.5GbE Controller [10ec:8125] (rev 05)
	Subsystem: Device [1f4c:b002]
	Kernel driver in use: r8169
	Kernel modules: r8169
02:00.0 Network controller [0280]: Intel Corporation Wi-Fi 6 AX210/AX211/AX411 160MHz [8086:2725] (rev 1a)
	Subsystem: Rivet Networks Wi-Fi 6 AX210/AX211/AX411 160MHz [1a56:1674]
	Kernel driver in use: iwlwifi
	Kernel modules: iwlwifi
03:00.0 Non-Volatile memory controller [0108]: Kingston Technology Company, Inc. Device [2646:501b]
	Subsystem: Kingston Technology Company, Inc. Device [2646:501b]
	Kernel driver in use: nvme
	Kernel modules: nvme
c4:00.0 VGA compatible controller [0300]: Advanced Micro Devices, Inc. [AMD/ATI] Phoenix1 [1002:15bf] (rev c1)
	Subsystem: Device [1f4c:b002]
	Kernel driver in use: vfio-pci
	Kernel modules: amdgpu
c4:00.1 Audio device [0403]: Advanced Micro Devices, Inc. [AMD/ATI] Rembrandt Radeon High Definition Audio Controller [1002:1640]
	Subsystem: Device [1f4c:b002]
	Kernel driver in use: vfio-pci
	Kernel modules: snd_hda_intel
c4:00.2 Encryption controller [1080]: Advanced Micro Devices, Inc. [AMD] Device [1022:15c7]
	Subsystem: Device [1f4c:b002]
	Kernel driver in use: ccp
	Kernel modules: ccp
c4:00.3 USB controller [0c03]: Advanced Micro Devices, Inc. [AMD] Device [1022:15b9]
	Subsystem: Device [1f4c:b002]
	Kernel driver in use: xhci_hcd
	Kernel modules: xhci_pci
c4:00.4 USB controller [0c03]: Advanced Micro Devices, Inc. [AMD] Device [1022:15ba]
	Subsystem: Device [1f4c:b002]
	Kernel driver in use: xhci_hcd
	Kernel modules: xhci_pci
c4:00.5 Multimedia controller [0480]: Advanced Micro Devices, Inc. [AMD] ACP/ACP3X/ACP6x Audio Coprocessor [1022:15e2] (rev 63)
	Subsystem: Device [1f4c:b002]
	Kernel driver in use: snd_pci_ps
	Kernel modules: snd_pci_acp3x, snd_rn_pci_acp3x, snd_pci_acp5x, snd_pci_acp6x, snd_acp_pci, snd_rpl_pci_acp6x, snd_pci_ps, snd_sof_amd_renoir, snd_sof_amd_rembrandt
c4:00.6 Audio device [0403]: Advanced Micro Devices, Inc. [AMD] Family 17h/19h HD Audio Controller [1022:15e3]
	DeviceName: Realtek ALC256
	Subsystem: Device [1f4c:b002]
	Kernel driver in use: snd_hda_intel
	Kernel modules: snd_hda_intel
c5:00.0 Non-Essential Instrumentation [1300]: Advanced Micro Devices, Inc. [AMD] Device [1022:14ec]
	Subsystem: Device [1f4c:b002]
c5:00.1 Signal processing controller [1180]: Advanced Micro Devices, Inc. [AMD] Device [1022:1502]
	Subsystem: Device [1f4c:b002]
c6:00.0 Non-Essential Instrumentation [1300]: Advanced Micro Devices, Inc. [AMD] Device [1022:14ec]
	Subsystem: Device [1f4c:b002]
c6:00.3 USB controller [0c03]: Advanced Micro Devices, Inc. [AMD] Device [1022:15c0]
	Subsystem: Device [1f4c:b002]
	Kernel driver in use: xhci_hcd
	Kernel modules: xhci_pci
c6:00.4 USB controller [0c03]: Advanced Micro Devices, Inc. [AMD] Device [1022:15c1]
	Subsystem: Device [1f4c:b002]
	Kernel driver in use: xhci_hcd
	Kernel modules: xhci_pci
c6:00.5 USB controller [0c03]: Advanced Micro Devices, Inc. [AMD] Device [1022:1668]
	Subsystem: Device [1f4c:b002]
	Kernel driver in use: thunderbolt
	Kernel modules: thunderbolt
c6:00.6 USB controller [0c03]: Advanced Micro Devices, Inc. [AMD] Device [1022:1669]
	Subsystem: Device [1f4c:b002]
	Kernel driver in use: thunderbolt
	Kernel modules: thunderbolt
root@pve:~# cat /etc/default/grub
# If you change this file, run 'update-grub' afterwards to update
# /boot/grub/grub.cfg.
# For full documentation of the options in this file, see:
#   info -f grub -n 'Simple configuration'

GRUB_DEFAULT=0
GRUB_TIMEOUT=5
GRUB_DISTRIBUTOR=`lsb_release -i -s 2> /dev/null || echo Debian`
GRUB_CMDLINE_LINUX_DEFAULT="quiet iommu=pt"
GRUB_CMDLINE_LINUX=""

# If your computer has multiple operating systems installed, then you
# probably want to run os-prober. However, if your computer is a host
# for guest OSes installed via LVM or raw disk devices, running
# os-prober can cause damage to those guest OSes as it mounts
# filesystems to look for things.
#GRUB_DISABLE_OS_PROBER=false

# Uncomment to enable BadRAM filtering, modify to suit your needs
# This works with Linux (no patch required) and with any kernel that obtains
# the memory map information from GRUB (GNU Mach, kernel of FreeBSD ...)
#GRUB_BADRAM="0x01234567,0xfefefefe,0x89abcdef,0xefefefef"

# Uncomment to disable graphical terminal
#GRUB_TERMINAL=console

# The resolution used on graphical terminal
# note that you can use only modes which your graphic card supports via VBE
# you can see them in real GRUB with the command `vbeinfo'
#GRUB_GFXMODE=640x480

# Uncomment if you don't want GRUB to pass "root=UUID=xxx" parameter to Linux
#GRUB_DISABLE_LINUX_UUID=true

# Uncomment to disable generation of recovery mode menu entries
#GRUB_DISABLE_RECOVERY="true"

# Uncomment to get a beep at grub start
#GRUB_INIT_TUNE="480 440 1"
root@pve:~# cat /etc/modprobe.d/vfio.conf
options vfio-pci ids=1002:15bf,1002:1640
softdep radeon pre: vfio-pci
softdep amdgpu pre: vfio-pci
softdep snd_hda_intel pre: vfio-pci
root@pve:~# cat /etc/modprobe.d/pve-blacklist.conf
# This file contains a list of modules which are not supported by Proxmox VE

# nvidiafb see bugreport https://bugzilla.proxmox.com/show_bug.cgi?id=701
blacklist nvidiafb
```

Regarding point 2, I have forgot to mention that I have already tried to install Win 10 and follow the guide. But also here without success.
On the win10 machine if I try to set the Hardware Display option to none, the system does not boot.

Best

**Reply by isc30** [reply]
Interesting, if `win10` with `SeaBIOS` it doesn't boot either, its probably a vbios issue. I can see that the `vfio-pci` driver is loaded properly for the 2 devices (GPU+Audio).

Can you try to recreate the Win10 VM so we can check for specific errors there?

> On the win10 machine if I try to set the Hardware Display option to none, the system does not boot.

  When using the virtual display adapter, if you open Device Manager, does the GPU show up? Were you able to install the drivers properly when not using it as Primary GPU?

**Reply by MMaiero** [reply]
Sure, this is my configuration for the Win 10 machine created today:

![image](https://github.com/isc30/ryzen-7000-series-proxmox/assets/10448126/175939fc-aa31-41d4-b2cb-68099b4fcb23)

When started the memory jumps to 90% and the CPU to 25% and stays there without really booting.

![image](https://github.com/isc30/ryzen-7000-series-proxmox/assets/10448126/bd1c5503-e03c-48ef-80e0-8d84fa6041a6)

If I force the stop of the VM and set the VirtIO GPU, the system boots

![image](https://github.com/isc30/ryzen-7000-series-proxmox/assets/10448126/2c14f6d5-c716-4443-8c9d-274a2be8c439)

**Reply by isc30** [reply, marked as answer]
From the message `The device cannot find enough free resources that it can use. Error code 12` it seems like it needs more RAM.
When a PCI device is virtualized like the GPU, it needs to have enough ram to allocate all the GPU RAM in the normal RAM. By default I think its 2gb, which it doesnt have spare. Try allocating 16gb to the Windows 10 machine and lets see what happens

**Reply by MMaiero** [reply]
Tried setting 16GB. When the display setting is set to none, same issue as before. The VM remains stuck as follows:

![image](https://github.com/isc30/ryzen-7000-series-proxmox/assets/10448126/a7180d49-45bd-4ece-b62e-78299f274b6a)

When the VirtIO GPU is set, the VM starts but it seems to lead to the same result:

![image](https://github.com/isc30/ryzen-7000-series-proxmox/assets/10448126/022fe3bc-180b-4230-a83a-c61bd0b0b553)

![image](https://github.com/isc30/ryzen-7000-series-proxmox/assets/10448126/453b28ac-7a42-45ed-be3f-be2a7c2d3b3d)

**Reply by isc30** [reply]
try changing `cpu` to the default one instead of `host`, and remove the passthrough USB stuff to make sure this PCI device is the only one being sent.

You can also check the iommu groups with this script, please send the output:
```bash
pvesh get /nodes/{nodename}/hardware/pci --pci-class-blacklist ""
```

**Reply by MMaiero** [reply]
It works!
I have tested your suggestion on setting the 16GB in the Win11 VM and actually that made the trick:

![image](https://github.com/isc30/ryzen-7000-series-proxmox/assets/10448126/2b0e8731-1e95-4f6b-9948-6442054706b3)

![image](https://github.com/isc30/ryzen-7000-series-proxmox/assets/10448126/af9154a2-1435-4d20-8f78-cbeca8c4cbd3)

The passthrough of the USB device can also be added or removed without effects.

Thanks again for all the support!

**Reply by isc30** [reply]
@MMaiero thats great! Can you do me a favour and upload your vbios to the repo? this way other people with your processor can have access to it :)

**Comment by isc30** [comment]
thanks a lot for all the feedback, ive updated the guide with a minimum of 12gb of RAM

**Comment by TKtheDEV** [comment]
Thank you both for the great documentation of making it work in first place and debugging for the UM790Pro!
Since I am trying to get my UM790Pro to work aswell, may I ask which Minisforum UEFI Version and Settings you are using (especially iGPU settings) @MMaiero?
~Tobi

**Reply by TKtheDEV** [reply]
Hi all, after setting up multiple VMs and trying all over again I cant get it to work still. 
![monitor](https://github.com/isc30/ryzen-7000-series-proxmox/assets/46247441/d35bbe0f-50c3-416d-9387-f48e53483810)
This is what my monitor shows me which indicates, that the GPU is initialized by the host as far as i know. The whole PC keeps crashing upon starting the VM. This is what my VM options look like:
![vmconf](https://github.com/isc30/ryzen-7000-series-proxmox/assets/46247441/49353a56-e80f-42e4-9b82-5a87ac42b5f7)
Thanks alot for taking the time to help me!

**Reply by isc30** [reply]
Hey I can recommend going step by step to see what could be the issue. Can you first try with Win10 + SeaBios and check if that works pls? @TKtheDEV

**Reply by TKtheDEV** [reply]
Hey @isc30
sorry for the late reply!
using seabios my whole host is crashing when starting the VM. I only get a VGA output using OVMF...
But after a host restart i first get a bluescreen inside the VM as you can see below and afterwards some error that my GPU driver cant be loaded...

this is my VM log
TASK ERROR: start failed: command '/usr/bin/kvm -id 209 -name 'win10,debug-threads=on' -no-shutdown -chardev 'socket,id=qmp,path=/var/run/qemu-server/209.qmp,server=on,wait=off' -mon 'chardev=qmp,mode=control' -chardev 'socket,id=qmp-event,path=/var/run/qmeventd.sock,reconnect=5' -mon 'chardev=qmp-event,mode=control' -pidfile /var/run/qemu-server/209.pid -daemonize -smbios 'type=1,uuid=dc9c60f3-2fb4-4890-adbc-5a2d3b81b133' -smp '6,sockets=1,cores=6,maxcpus=6' -nodefaults -boot 'menu=on,strict=on,reboot-timeout=1000,splash=/usr/share/qemu-server/bootsplash.jpg' -vga none -nographic -cpu 'host,hv_ipi,hv_relaxed,hv_reset,hv_runtime,hv_spinlocks=0x1fff,hv_stimer,hv_synic,hv_time,hv_vapic,hv_vpindex,+kvm_pv_eoi,+kvm_pv_unhalt' -m 12288 -readconfig /usr/share/qemu-server/pve-q35-4.0.cfg -device 'vmgenid,guid=050b0ca5-f1df-413f-8014-5e147f0835c4' -device 'usb-tablet,id=tablet,bus=ehci.0,port=1' -device 'vfio-pci,host=0000:c4:00.0,id=hostpci0,bus=ich9-pcie-port-1,addr=0x0,romfile=/usr/share/kvm/vbios_7940hs.bin' -device 'vfio-pci,host=0000:c4:00.1,id=hostpci1,bus=ich9-pcie-port-2,addr=0x0' -chardev 'socket,path=/var/run/qemu-server/209.qga,server=on,wait=off,id=qga0' -device 'virtio-serial,id=qga0,bus=pci.0,addr=0x8' -device 'virtserialport,chardev=qga0,name=org.qemu.guest_agent.0' -device 'virtio-balloon-pci,id=balloon0,bus=pci.0,addr=0x3,free-page-reporting=on' -iscsi 'initiator-name=iqn.1993-08.org.debian:01:76a2a3c4a6a4' -drive 'file=/var/lib/vz/template/iso/virtio-win-0.1.240.iso,if=none,id=drive-ide1,media=cdrom,aio=io_uring' -device 'ide-cd,bus=ide.2,unit=0,drive=drive-ide1,id=ide1,bootindex=103' -drive 'file=/var/lib/vz/template/iso/Windows10Pro.iso,if=none,id=drive-ide2,media=cdrom,aio=io_uring' -device 'ide-cd,bus=ide.1,unit=0,drive=drive-ide2,id=ide2,bootindex=101' -device 'virtio-scsi-pci,id=scsihw0,bus=pci.0,addr=0x5' -drive 'file=/var/lib/vz/images/209/vm-209-disk-1.qcow2,if=none,id=drive-scsi0,discard=on,format=qcow2,cache=none,aio=io_uring,detect-zeroes=unmap' -device 'scsi-hd,bus=scsihw0.0,channel=0,scsi-id=0,lun=0,drive=drive-scsi0,id=scsi0,bootindex=100' -netdev 'type=tap,id=net0,ifname=tap209i0,script=/var/lib/qemu-server/pve-bridge,downscript=/var/lib/qemu-server/pve-bridgedown,vhost=on' -device 'virtio-net-pci,mac=C2:98:5A:8A:3A:59,netdev=net0,bus=pci.0,addr=0x12,id=net0,rx_queue_size=1024,tx_queue_size=256,bootindex=102' -rtc 'driftfix=slew,base=localtime' -machine 'hpet=off,smm=off,type=pc-q35-8.0+pve0' -global 'kvm-pit.lost_tick_policy=discard'' failed: got timeout

**Comment by MMaiero** [comment]
> Hi all, after setting up multiple VMs and trying all over again I cant get it to work still. ![monitor](https://user-images.githubusercontent.com/46247441/278836555-d35bbe0f-50c3-416d-9387-f48e53483810.jpg) This is what my monitor shows me which indicates, that the GPU is initialized by the host as far as i know. The whole PC keeps crashing upon starting the VM. This is what my VM options look like: ![vmconf](https://user-images.githubusercontent.com/46247441/278836769-49353a56-e80f-42e4-9b82-5a87ac42b5f7.png) Thanks alot for taking the time to help me!

Hi, in terms of monitor displaying, I see the same in my case. Once the Vm is started it then picks the monitor up and starts the displaying. So, it is not a direct indicator of an issue there.
I would check step by step if the guide has been properly applied.

Also, what version of the BIOS are you running? I am currently on 1.07.

**Reply by TKtheDEV** [reply]
Hi again,
My Bios is 1.07 aswell.
got the monitor output working now using UEFI on a new VM with the following config, but I am having trouble with the GPU driver. Had a bluescreen upon boot and the yellow exclamation mark in device manager... showing code 43
![vmconf](https://github.com/isc30/ryzen-7000-series-proxmox/assets/46247441/63536016-1df3-4298-9bdf-2da57648513e)
![startup](https://github.com/isc30/ryzen-7000-series-proxmox/assets/46247441/2d89da0b-953a-44c2-b6c4-5185622b1eb1)
![bluescreen](https://github.com/isc30/ryzen-7000-series-proxmox/assets/46247441/1e3b47c8-7b86-4190-8aa6-0977b61ca4d4)
![devicemanager](https://github.com/isc30/ryzen-7000-series-proxmox/assets/46247441/ca2b9e32-3e8a-486f-958b-93b50582fb62)

**Reply by TKtheDEV** [reply]
This is what I get now. are you able to change display resolution and use the GPU? mine is stuck on windows basic DA driver fsr

**Comment by rayd0n** [comment]
Hi @isc30,

I am struggling as well to get it up and running on an UM790 pro. I have been following your documentation step by step. 

I am running the bios 1.07 and proxmox 8.0.3.

Below is how I have configured the proxmox host system.

```
root@proxmox:~# cat /etc/default/grub | grep ^GRUB_CMD
GRUB_CMDLINE_LINUX_DEFAULT="quiet amd_iommu=on iommu=pt"
```

```
root@proxmox:~# cat /etc/modules
# /etc/modules: kernel modules to load at boot time.
#
# This file contains the names of kernel modules that should be loaded
# at boot time, one per line. Lines beginning with "#" are ignored.
# Parameters can be specified after the module name.

vfio
vfio_iommu_type1
vfio_pci
vfio_virqfd
```

```
root@proxmox:~# cat /etc/modprobe.d/vfio.conf 
options vfio-pci ids=1002:15bf,1002:1640
softdep radeon pre: vfio-pci
softdep amdgpu pre: vfio-pci
softdep snd_hda_intel pre: vfio-pci
```

```
root@proxmox:~# lspci -nnk
c4:00.0 VGA compatible controller [0300]: Advanced Micro Devices, Inc. [AMD/ATI] Phoenix1 [1002:15bf] (rev c1)
	Subsystem: Device [1f4c:b002]
	Kernel driver in use: vfio-pci
	Kernel modules: amdgpu
c4:00.1 Audio device [0403]: Advanced Micro Devices, Inc. [AMD/ATI] Rembrandt Radeon High Definition Audio Controller [1002:1640]
	Subsystem: Device [1f4c:b002]
	Kernel driver in use: vfio-pci
	Kernel modules: snd_hda_intel
```

I have also tried with Win10 and SeaBios but it does not work either. I have tried both Win10 and Win11 with UEFI bios. When I setup the display to "none", most of the times the VM won't even fully boot up. I am unable to connect via RDP. If I change the display to "default", I will get into the machine most of the times, but the card is not activated (error 43).

On very few occasions I managed to get the card loaded, but when I did, graphics performed was just terrible. When I went to https://webglsamples.org/ I would maybe get 3/4 fps, so it does seem that passthrough was not working properly either.

Below is what my VM setup currently looks like.

<img width="671" alt="Screenshot 2023-10-30 at 16 35 31" src="https://github.com/isc30/ryzen-7000-series-proxmox/assets/47508/8acc7501-4998-41b8-8426-12ca2b1a2fd1">

Any suggestions on what I could have done wrong?

**Reply by rayd0n** [reply]
After rebooting promox I just happened to get the card loaded properly in the system. As you can see below it's activated but graphics performance is really really terrible. The whole machine is on it's last leg rendering the test graphics.

<img width="227" alt="Screenshot 2023-10-30 at 16 48 04" src="https://github.com/isc30/ryzen-7000-series-proxmox/assets/47508/4c4e49a1-7e2f-4281-b108-96fd21027112">

<img width="293" alt="Screenshot 2023-10-30 at 16 47 18" src="https://github.com/isc30/ryzen-7000-series-proxmox/assets/47508/10906b6d-4e92-468b-a8a3-11c253398225">

<img width="401" alt="Screenshot 2023-10-30 at 16 46 59" src="https://github.com/isc30/ryzen-7000-series-proxmox/assets/47508/798f6257-e803-46fc-b930-59cfa31d581a">

**Comment by TKtheDEV** [comment]
Yo everyone, Tobi here,
after spending ~120hrs I finally made it work on my Minisforum UM790PRO using the following:
-Windows 11
-Minisforum UEFI v1.07
-Proxmox 8.04
-Kernel 6.2.16-15-pve

![VMsettings](https://github.com/isc30/ryzen-7000-series-proxmox/assets/46247441/e13f8272-8600-4dcb-aec2-5bd26a515667)

You may set Display to none aswell, just be aware of the AMD Reset Bug. I recommend setting it to Default VGA during reboots (e.g. for Windows Update)

![VMconf](https://github.com/isc30/ryzen-7000-series-proxmox/assets/46247441/36a29c52-0774-4ed6-99b7-943fc4500c5f)

!!!The "args" line was important to make it work for me!!! (and sorry for it being an image)

![dvmgr](https://github.com/isc30/ryzen-7000-series-proxmox/assets/46247441/a5605acf-fe88-40fd-a053-5ce6441bbd82)
This is what the final result looks like (DX12_2 Raytracing etc working)

Reset Bug is still happening and couldn't get Riot Vanguard to work yet either. I'll post once I got that working!
The "romfiles" im using are the ones from the main repo

Thanks alot again @MMaiero and  @isc30 for your help, really appreciate it!

~Tobi

**Reply by isc30** [reply]
Hey that's great! Do you mind creating a PR with the args? They won't do anything for other CPUs like 7335hs but it helps for others like yours

**Reply by isc30** [reply]
done it myself, can you check if the current guide works?

**Reply by TKtheDEV** [reply]
Yes I will as soon as i am at home again, will post the follow up here

**Reply by Seemops1337** [reply]
Hey Tobi, did u find a fix 4 the Reset bug on the UM790Pro? :)
My VM is also working just the Reset Bug is left...

**Reply by TKtheDEV** [reply]
Hi @Seemops1337,
I gave up on windows entirely. The reason I wanted to have proper GPU Accel was to run jellyfin and do some gaming. But I finally was able to get a replacement track pad for my laptop that works with Linux and got rid of windows entirely. On the server side I switched to LXC containers with GPU access, since many LXCs can access the GPU simultaneously that way (kinda similar to sr-iov for VMs). 
Have you tried the latest bios? And btw if you flash it make sure you use the command on their website. I bricked mine when using a generic command and had to hardware flash it afterwards :/ works again now tho
